web/libs/editor/src/regions/Result.js

```diff
@@ -201,13 +219,21 @@ const Result = types
         return true;
       };
 
-      if (control.visiblewhen === "choice-selected") {
+      // When perregion is used, we must ignore the visibility of the components and focus only on the selection
+      if (control.perregion && control.visiblewhen === "choice-selected") {
         return isChoiceSelected();
       }
+
       if (control.visiblewhen === "choice-unselected") {
         return !isChoiceSelected();
       }
 
+      // We need to check if there is any node up in the tree with visibility restrictions so we can determine
+      // if the element is selected considering its own visibility
+      if (findParentWithVisibleWhen(control, "choice-selected")) {
```

Comment:
I'd make this fix only for non-per-region tags explicitly stating it here. Right now this condition is not fully opposite to the previous one, so we might have per-region tag without visibility params but within the `View` with `visibleWhen="choice-selected"` and this can lead to errors.
```suggestion
      if (!control.perregion && findParentWithVisibleWhen(control, "choice-selected")) {
```

Reply:
Added